devito/ops/types.py

```diff
@@ -17,6 +19,39 @@ def _C_typedata(self):
         return super()._C_typedata
 
 
+class RawAccessible(basic.Symbol):
+    is_Scalar = True
+
+    def __init_finalize__(self, name, read_only=True, **kwargs):
+        self.read_only = read_only
+        super().__init_finalize__(name, **kwargs)
+
+    @property
+    def _C_typename(self):
+        # TODO: can't change self.dtype from __init_finalize__
+        return '%s%s *' % (
+            'const ' if self.read_only else '',
+            ctypes_to_cstr(c_float)
+        )
+
+
+class RawAccess(basic.Basic, sympy.Basic):
+    def __init__(self, base, *args, **kwargs):
+        self.base = base
+        super().__init__(*args, **kwargs)
+
+    def __str__(self):
+        return '(*%s)' % self.base._C_name
```

Comment:
I just thought that a nicer way would be to introduce a SymPy subclass along the lines of https://github.com/opesci/devito/blob/master/devito/symbolics/extended_sympy.py#L286

this way you wouldn't need to introduce a new symbolic type. And you would be able to use that new class  to wrap any symbolic object you like, potentially even a whole function call.... `(*a)`, `(**a)` (with a suitable parameter `nindirections`), `(*foo())`, ...